playground/backend/internal/executors/executor.go

```diff
@@ -13,19 +13,102 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Interface for all executors (Java/Python/Go/SCIO)
+// Package executors
 package executors
 
-type executor interface {
-	// Validate validates executable file.
-	// Return result of validation (true/false) and error if it occurs
-	Validate(filePath string) (bool, error)
+import (
+	pb "beam.apache.org/playground/backend/internal/api"
+	"fmt"
+	"os/exec"
+)
 
-	// Compile compiles executable file.
-	// Return error if it occurs
-	Compile(filePath string) error
+type validatorWithArgs struct {
+	validator func(filePath string, args ...interface{}) error
+	args      []interface{}
+}
+
+type preparationWithArgs struct {
+	prepare func(filePath string, args ...interface{}) error
+	args    []interface{}
+}
```

Comment:
Is it possible to combine `validatorWithArgs` and `preparationWithArgs` into one generic structure like `funcitonWithArgs`?

Reply:
Done.